UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/GasMix.cs

```diff
@@ -438,6 +438,24 @@ public void MergeGasMixes(List<PipeData> otherGas)
 			}
 		}
 
+		public GasSO GetBiggestGasSOInMix()
+		{
+			if(GasData.GasesArray.Count == 0)
+			{
+				return null;
+			}
+
+			GasSO bigGas = GasData.GasesArray[0].GasSO;
+			float lastBigNumber = 0f;
+			foreach (var gas in GasData.GasesArray)
+			{
+				if (gas.Moles < lastBigNumber) continue;
+				bigGas = gas.GasSO;
+				lastBigNumber = gas.Moles;
```

Comment:
Not sure why you want `lastBigNumber` it works fine without it, and instead just checking in the IF statement like before

Reply:
because a gasSO does not hold moles
it's static data that tells the game what type of gas it is and what are it's properties.
the lastbignumber variable is there for us to compare the largest number of moles in gasses in that array.

Reply:
ah sorry, i was thinking it was GasMix. Cool